File: src/main/java/com/freya02/botcommands/internal/events/EventListenerParameter.java
package com.freya02.botcommands.internal.events;

import com.freya02.botcommands.internal.application.CommandParameter;

import java.lang.annotation.Annotation;
import java.lang.reflect.Parameter;
import java.util.List;

public class EventListenerParameter extends CommandParameter<Object> {
	public EventListenerParameter(Parameter parameter, int index) {
		super(null, parameter, index);
	}

	@Override
	protected List<Class<? extends Annotation>> getOptionAnnotations() {
		return List.of();
	}
}

File: src/main/java/com/freya02/botcommands/internal/events/EventListenerParameter.kt
package com.freya02.botcommands.internal.events

import com.freya02.botcommands.internal.application.CommandParameter
import kotlin.reflect.KClass
import kotlin.reflect.KParameter

class EventListenerParameter(parameter: KParameter, index: Int) : CommandParameter<Any>(null, parameter, index) {
    override val optionAnnotations: List<KClass<out Annotation>>
        get() = listOf()
}